ping/rust/src/lib.rs

```diff
@@ -70,19 +73,20 @@ where
         // two peers dialling each other at the same time).
         //
         // We can do this because sync service pubsub is ordered.
-        .take_while(|a| ready(a != &local_addr));
+        .take_while(|a| ready(a != &local_addr && a != &dialable_multiaddr));
```

Comment:
I think we can now simplify this.

```suggestion
        .take_while(|a| ready(a != &dialable_multiaddr));
```